src/components/accession2/viabilityTesting/NewViabilityTestModal.tsx

```diff
@@ -97,6 +100,10 @@ export default function NewViabilityTestModal(props: NewViabilityTestModalProps)
       if (response.requestSucceeded) {
         reload();
         onCloseHandler();
+        if (testCompleted) {
+          setSavedRecord({ ...record });
+          setOpenViabilityResultModal(true);
+        }
```

Comment:
if user opens an already completed test and only edits the notes, we may still end up calling this. maybe this should be `if (testCompleted && !readOnly)`

Reply:
Good catch!